### PR TITLE
Convert PayPal module classes to Kotlin

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes

 - Convert PayPalClient, PayPalInternalClient, and PayPalLineItem to Kotlin.
 - This completes the Kotlin conversion for all of the PayPal module classes.

### Checklist

 - [X] Added a changelog entry
 - [ ] Relevant test coverage

### CODE REVIEW

File: PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalLineItemKind.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+/**
+ * The type of PayPal line item.
+ *
+ * [KIND_CREDIT] A line item that is a credit.
+ * [KIND_DEBIT] A line item that debits.
+ */
+enum class PayPalLineItemKind(val stringValue: String) {
+    KIND_CREDIT("credit"),
+    KIND_DEBIT("debit")
+}
```
Comment (jaxdesmarais) [inline PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalLineItemKind.kt:12]: Is duplicating the word "kind" generally a best practice in Kotlin/Java or could we drop it from either the class name or enum name?

**Review by jaxdesmarais — APPROVED** [review]

File: PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+            sendVaultRequest(context, payPalRequest, callback)
+        }
+    }
+
+    private fun sendCheckoutRequest(
+        context: Context,
+        payPalCheckoutRequest: PayPalCheckoutRequest,
+        callback: PayPalPaymentAuthCallback
+    ) {
+        braintreeClient.sendAnalyticsEvent("paypal.single-payment.selected", analyticsParams)
```
Comment (sarahkoop) [inline PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt:83]: These might have accidentally made it into v5 - I think all analytics events should be in the `PayPalAnalytics` enum now. Should these be removed?

Comment (tdchow) [reply PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt:83]: Good catch! Mind if I make this change in a follow-up PR? I feel we should reduce the amount of changes in the Kotlin conversion PR.

File: PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalLineItem.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ * @property url The URL to product information.
+ */
+
+@Parcelize
+data class PayPalLineItem
+@JvmOverloads constructor(
+    val kind: PayPalLineItemKind,
+    val name: String,
+    val quantity: String,
+    val unitAmount: String?,
```
Comment (sarahkoop) [inline PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalLineItem.kt:34]:
```suggestion
    val unitAmount: String,
```

I think this value is required

**Review by sarahkoop — APPROVED** [review]

File: PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalLineItem.kt (hunk 3 above)
Comment (tdchow) [reply PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalLineItem.kt]: Whoops, I removed the default value but kept the type as nullable. Just updated this.

File: PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt (hunk 2 above)
Comment (tdchow) [reply PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt:83]: Opened https://github.com/braintree/braintree_android/pull/1093

**Pull request merged**